extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+class PortableInfoboxErrorRenderService extends WikiaService {
+
+	const XML_DEBUG_TEMPLATE = 'PortableInfoboxMarkupDebug.mustache';
+	const XML_ERR_GENERAL = 'xml-parse-error';
+
+	private $errorList = [];
+	private $supportedErrors = [
+		5 => 'XML_ERR_DOCUMENT_END',
+		26 => 'XML_ERR_UNDECLARED_ENTITY',
+		39 => 'XML_ERR_ATTRIBUTE_NOT_STARTED',
+		41 => 'XML_ERR_ATTRIBUTE_WITHOUT_VALUE',
+		65 => 'XML_ERR_SPACE_REQUIRED',
+		68 => 'XML_ERR_NAME_REQUIRED',
+		73 => 'XML_ERR_GT_REQUIRED',
+		76 => 'XML_ERR_TAG_NAME_MISMATCH',
+		77 => 'XML_ERR_TAG_NOT_FINISHED'
+	];
+
+	/* @var $templateEngine Wikia\Template\MustacheEngine */
+	private $templateEngine;
+
+	function __construct( $errorList ) {
+		$this->templateEngine = ( new Wikia\Template\MustacheEngine )
+			->setPrefix( dirname( __FILE__ ) . '/../templates' );
+		$this->errorList = $errorList;
+	}
+
+	public function renderMarkupDebugView( $sourceCode ) {
```

Comment:
This method reads quite hard. It looks like this method could call we have:
* getLinesWithErrors($sourceCode)
** that could use something along the lines of getErrorsByLine($this->errorList)
and in the end handle rendering